cocos/2d/CCSpriteFrameCache.cpp

```diff
@@ -762,8 +762,12 @@ bool SpriteFrameCache::PlistFramesCache::eraseFrame(const std::string &frame)
     auto itFrame = _indexFrame2plist.find(frame);
     if (itFrame != _indexFrame2plist.end())
     {
-        _indexPlist2Frames[itFrame->second].erase(frame);   //update index plist->[frameNames]
+        auto plist = itFrame->second;
+        _indexPlist2Frames[plist].erase(frame);             //update index plist->[frameNames]
         _indexFrame2plist.erase(itFrame);                   //update index frame->plist
+        // erase plist index if all frames was erased
+        if (_indexFrame2plist.empty())
+            _indexPlist2Frames.erase(plist);
```

Comment:
if a plist file was added, and then all frames was erased, the plist index may still existed, like
```
(lldb) print this->_spriteFramesCache
(cocos2d::SpriteFrameCache::PlistFramesCache) $4 = {
  _spriteFrames = {
    _data = size=0 {}
  }
  _indexPlist2Frames = size=1 {
    [0] = {
      __cc = {
        first = "Images/test_polygon.plist"
        second = size=0 {}
      }
      __nc = {
        first = "Images/test_polygon.plist"
        second = size=0 {}
      }
    }
  }
  _indexFrame2plist = size=0 {}
}
``` 
so that frames will be prevent added when load same name plist  again

Reply:
it should be a bug caused by https://github.com/cocos2d/cocos2d-x/pull/19083 ,please review @PatriceJiang

Reply:
ok